From 2b42c620e7525134ad54bb3480e4832947de68ce Mon Sep 17 00:00:00 2001
From: Tian Yang <tiany@codeaurora.org>
Date: Mon, 28 Sep 2015 16:44:09 -0500
Subject: [PATCH] skbuff_debug: check for return code on activate

debug_oject_activate() can return an error, so let's check it
and print the error code

Change-Id: I8e08074df4314dd9c1c323c4d891d5ba39a63a12
Signed-off-by: Matthew McClintock <mmcclint@codeaurora.org>
Signed-off-by: Casey Chen <kexinc@codeaurora.org>
---
 net/core/skbuff_debug.c | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

--- a/net/core/skbuff_debug.c
+++ b/net/core/skbuff_debug.c
@@ -42,15 +42,19 @@ static struct debug_obj_descr skbuff_deb
 	.fixup_free	= skbuff_debugobj_fixup,
 };
 
-inline void skbuff_debugobj_init_and_activate(struct sk_buff *skb)
+inline void skbuff_debugobj_activate(struct sk_buff *skb)
 {
-	debug_object_init(skb, &skbuff_debug_descr);
-	debug_object_activate(skb, &skbuff_debug_descr);
+	int ret = debug_object_activate(skb, &skbuff_debug_descr);
+
+	if (ret)
+		WARN(1, "skb_debug: failed to activate err = %d skb = 0x%p\n",
+		     ret, skb);
 }
 
-inline void skbuff_debugobj_activate(struct sk_buff *skb)
+inline void skbuff_debugobj_init_and_activate(struct sk_buff *skb)
 {
-	debug_object_activate(skb, &skbuff_debug_descr);
+	debug_object_init(skb, &skbuff_debug_descr);
+	skbuff_debugobj_activate(skb);
 }
 
 inline void skbuff_debugobj_deactivate(struct sk_buff *skb)
